include/wlr/types/wlr_xdg_shell.h

```diff
@@ -296,9 +296,14 @@ uint32_t wlr_xdg_toplevel_set_tiled(struct wlr_xdg_surface *surface,
 		uint32_t tiled_edges);
 
 /**
- * Request that this xdg surface closes.
+ * Request that this xdg toplevel closes.
  */
-void wlr_xdg_surface_send_close(struct wlr_xdg_surface *surface);
+void wlr_xdg_toplevel_send_close(struct wlr_xdg_surface *surface);
+
+/**
+ * Request that this xdg popup closes.
+ **/
+void wlr_xdg_popup_destroy(struct wlr_xdg_surface *surface);
```

Comment:
Why isn't this named `wlr_xdg_popup_done/send_done`? And at least, because it takes a popup, why not take a `wlr_xdg_popup` as a parameter instead?

Reply:
>Why isn't this named wlr_xdg_popup_done/send_done?

Because it effectively _destroys_ the `wlr_xdg_popup`. I prefer to make it clear the reference is not valid anymore after calling this function.

>And at least, because it takes a popup, why not take a wlr_xdg_popup as a parameter instead?

This is a more general issue. A lot of functions in this header could be more type-safe by taking a toplevel/popup instead of a xdg_surface. I agree this should be fixed, but probably as a separate PR.

Reply:
> Because it effectively _destroys_ the `wlr_xdg_popup`. I prefer to make it clear the reference is not valid anymore after calling this function.
> 
> >

Well, it just sends a done event if I understand the code correctly. The protocol says "The client should destroy the xdg_popup object at this point." which I interpret that the popup might be "alive" for some time after this request. Feel free to just tell me I'm wrong if that's the case.

Reply:
You're right. However, I'd like to free the popup and make the resource inert in this case. That's what we try to do when the compositor decides to free a resource and notifies the client about it.

Reply:
The client _should_ destroy the resource. But making the resource inert doesn't mean we free it. Inert means the resource can no longer be used by the client (only the `destroy` request can be used) and we can free our state.

Reply:
Okay, maybe my understanding of the protocol language isn't good enough. I thought that a client may choose to keep its popup open (so this was more like a hint).